### PR TITLE
Adds CSS URL rewriting capability w/ tests

### PR DESCRIPTION
**Pull request by kara-ryli**
Similar to @evocateur's commit, but I used URIjs to handle a few more URL cases and added some unit tests to validate the change.

### CODE REVIEW

File: lib/combohandler.js [hunk 1]
```diff
@@ -26,6 +28,10 @@ exports.combine = function (config) {
     if (typeof maxAge === 'undefined') {
         maxAge = 31536000; // one year in seconds
     }
+    
+    if (basePath) {
+      basePath += "/";
```
Comment (rgrove) [inline lib/combohandler.js:33]: What happens if `basePath` already has a trailing slash?

File: lib/combohandler.js [hunk 2]
```diff
@@ -179,3 +186,13 @@ function parseQuery(url) {
 
     return parsed;
 }
+
+// rewrites URLs in CSS files
+function rewriteCSSURLs(base, path, data) {
+    return data.replace(/url\(\s*['"]?([^)'"]*)['"]?\s*\)/g, function (substr, match) {
```
Comment (rgrove) [inline lib/combohandler.js]: Might want to tweak this regex to make it just a bit more restrictive. I'd like to be extra sure we don't mangle anything unexpectedly: `/[\s:]url\(\s*(['"])?([^)'"]*)\1\s*\)/g`

The URL becomes the second subpattern match in this case, so the replacer function parameters will also need to change.

**Comment by rgrove** [comment]
Thanks for this!

I had been holding off on merging @evocateur's changes because I wanted to come up with some kind of middleware-like way of supporting optional functionality like this, but I just haven't had the time. Since this is something people have been asking for, I'll go ahead and merge it once my minor comments above have been addressed.

**Comment by kara-ryli** [comment]
How have I be developing all this time without realizing JavaScript `RegExp`s supported back references?

Also, it looks like URI's absoluteTo method automatically encodes naughty bits of URLs, so a test looking for:

```
background:url("\)\";'\(.png") -> background:url("/rewritten/\)\";'\(.png")
```

fails, because URI is turning that into `/%5C%29%5C%22;%27%5C%28.png`. Is forcing URI encoding OK for this scenario? Or is it more important to honor the original input?

**Comment by rgrove** [comment]
Hmm. I'd prefer to honor the original input. If it's broken, maybe we can just leave it untouched?

**Comment by kara-ryli** [comment]
I don't think it's broken--URI is just being opinionated. The two URLs are equivalent, the CSS syntax is valid, and the encoding is actually suggested by the CSS spec. Also, this only occurs with resolvable URLs and not URIs (e.g. data URIs).

What I wouldn't want to do is be inconsistent within a file, e.g.:

```
#no-quotes { background: url(/rewritten/no-quotes.png);}
#escaped-stuff { background:url("/rewritten/%5C%29%5C%22;%27%5C%28.png"); }
```

Seems superior to

```
#no-quotes { background: url(/rewritten/no-quotes.png);}
#escaped-stuff { background:url("\)\";'\(.png"); }
```

Because the former will still resolve to the correct location while the later will not.

**Comment by rgrove** [comment]
So encoding the `\` characters doesn't break the URL in the first case, even though they're only escape sequence delimiters and not actual URL characters? That seems odd.

**Comment by kara-ryli** [comment]
Ah...you're correct and I was mis-reading it.

**Comment by rgrove** [comment]
Thanks!

**Pull request merged**